verifiers/utils/sigstore_tuf.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"sync/atomic"
+
+	sigstoreTUF "github.com/sigstore/sigstore-go/pkg/tuf"
+)
+
+// cache the default Sigstore TUF client.
+var defaultSigstoreTUFClientAtomicValue atomic.Value
+
+// SigstoreTUFClient is the interface for the Sigstore TUF client.
+type SigstoreTUFClient interface {
+	// GetTarget retrieves the target file from the TUF repository.
+	GetTarget(target string) ([]byte, error)
+}
+
+// GetDefaultSigstoreTUFClient returns the default Sigstore TUF client.
+// The client will be cacehd in memory.
+func GetDefaultSigstoreTUFClient() (SigstoreTUFClient, error) {
+	value := defaultSigstoreTUFClientAtomicValue.Load()
+	if value != nil {
+		return value.(SigstoreTUFClient), nil
+	}
+	sigstoreTUFClient, err := sigstoreTUF.DefaultClient()
+	if err != nil {
+		return nil, err
+	}
+	defaultSigstoreTUFClientAtomicValue.Store(sigstoreTUFClient)
```

Comment:
Do you want an atomic or https://pkg.go.dev/sync#Once? Before the TUF client is initialized, if you had multiple threads calling `GetDefaultSigstoreTUFClient`, then `Load` may return `nil` for each thread, and multiple clients will be created.

Reply:
Thanks, I think this is a nicer way